src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -294,23 +280,25 @@ def recursive_python_sweep(
             return self.play_sequence_in_sweep_recursion(qubits, sequence, or_sequence, execution_parameters)
 
         if not self.get_if_python_sweep(sequence, *sweepers):
-            toti, totq = self._execute_sweeps(sequence, qubits, sweepers, average)
+            toti, totq = self._execute_sweeps(sequence, qubits, sweepers)
             res = self.convert_sweep_results(or_sequence, qubits, toti, totq, execution_parameters)
             return res
 
         sweeper = sweepers[0]
         values = []
         for idx, _ in enumerate(sweeper.indexes):
             val = np.linspace(sweeper.starts[idx], sweeper.stops[idx], sweeper.expts)
+            if sweeper.parameters[idx] in rfsoc.Parameter.variants({"duration", "delay"}):
+                val = val.astype(int)
             values.append(val)
 
-        results = {}
+        results: dict[str, Union[IntegratedResults, SampleResults]] = {}
         for idx in range(sweeper.expts):
             # update values
             for jdx, kdx in enumerate(sweeper.indexes):
                 sweeper_parameter = sweeper.parameters[jdx]
                 if sweeper_parameter is rfsoc.Parameter.BIAS:
-                    qubits[kdx].flux.bias = values[jdx][idx]
+                    qubits[list(qubits.keys())[kdx]].flux.offset = values[jdx][idx]
```

Comment:
```suggestion
                    qubits[list(qubits)[kdx]].flux.offset = values[jdx][idx]
```
You can remove `.keys()` since calling `list` on a dictionary returns the list of keys.